research/query_service/gremlin/scripts/run_gaia.py

```diff
@@ -0,0 +1,247 @@
+        if args.download:
+            download_from_hdfs()
+        # After downloading data to local
+        local_dir = tmp_dir
+
+    program = os.path.join(os.getcwd(), "bin", "par_loader")
+    pool = []
+    env = "RUST_LOG=Info"
+    if args.host_file is not None:
+        host_list_file = os.path.join(tmp_dir, "hosts")
+        for i in range(num_machines):
+            log_file = open(os.path.join(log_dir, "%02d.log" % i), 'w')
+            run_params = "%s %s %s %s %d -w %d -n %d -p %d -h %s" % \
+                         (env, program, local_dir, args.graph_dir,
+                          args.raw_partitions, args.workers, num_machines, i, host_list_file)
+            cmd = ["ssh", hosts[i][0], run_params]
+            print(cmd)
+            pool.append(subprocess.Popen(cmd, stdout=log_file, stderr=log_file))
+            log_file.close()
+        sync_process(pool)
+    else:
+        cmd = "%s %s %s %s %d -w %d" % \
+                     (env, program, local_dir, args.graph_dir, args.raw_partitions, args.workers)
+        print(cmd)
+        os.system(cmd)
+
+
+def start_rpc():
+    """
+    Run RPC service
+    """
+    # Check the required arguments for starting RPC service
+    if args.graph_dir is None:
+        print("Must specify -g/--graph_dir")
+        parser.print_help()
+        exit(1)
+
+    graph_name = extract_file_name(args.graph_dir)[1]
+    log_dir = os.path.join(os.getcwd(), "logs", "start_rpc", "%s_m%d" % (graph_name, num_machines))
+    if not os.path.exists(log_dir):
+        os.makedirs(log_dir)
+
+    program = os.path.join(os.getcwd(), "bin", "start_rpc_server")
+    pool = []
+    env = "RUST_LOG=graph_store=Info,gremlin_core=Info,pegasus_server=Info DATA_PATH=\"%s\"" % args.graph_dir
+    if args.host_file is not None:
+        prev_host = ""
+        prev_port = 0
+        for i in range(num_machines):
+            log_file = open(os.path.join(log_dir, "%02d.log" % i), 'w')
+            env_i = "%s PARTITION_ID=%d" % (env, i)
+            run_params = "%s %s -h %s -i %d" % (env_i, program, args.host_file, i)
+            port_opt = "1234"
+            if args.port is not None:
+                port_opt = args.port
+            if prev_port == 0:
+                prev_port = int(port_opt)
+                port = prev_port
+            if hosts[i][0] == prev_host:
+                # Running in the same host, the port can not be the same
+                # For now, we simply increment the port number by 1 for the following hosts
+                port = prev_port + 1
+            prev_port = port
+            run_params = "%s -p %d --report" % (run_params, port)
+            cmd = ["ssh", hosts[i][0], run_params]
+            print(cmd)
+            pool.append(subprocess.Popen(cmd, stdout=log_file, stderr=log_file))
+            prev_host = hosts[i][0]
+            log_file.close()
+        sync_process(pool)
+    else:
+        cmd = "%s %s" % (env, program)
+        print(cmd)
+        os.system(cmd)
```

Comment:
ditto

Reply:
I removed these codes. But there are other issues like "partial exectuble path."

Reply:
Address the problem with:

```python
cmd = ["cmake", "."]  # not ok

cmd = [shutil.which("cmake"), "."] # is ok
```

Reply:
Nice, it is okay now. Thank you @lidongze0629